sdk/storage/Azure.Storage.Blobs.AspNetCore.DataProtection/src/AzureStorageBlobDataProtectionBuilderExtensions.cs

```diff
@@ -0,0 +1,125 @@
+using Azure.Storage.Blobs.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.DataProtection.KeyManagement;
+using Microsoft.Extensions.DependencyInjection;
+
+#pragma warning disable AZC0001 // Extension methods have to be in the correct namespace to appear in intellisense.
+namespace Microsoft.AspNetCore.DataProtection
+#pragma warning disable
+{
+    /// <summary>
+    /// Contains Azure-specific extension methods for modifying a
+    /// <see cref="IDataProtectionBuilder"/>.
+    /// </summary>
+    public static class AzureStorageBlobDataProtectionBuilderExtensions
+    {
+        /// <summary>
+        /// Configures the data protection system to persist keys to the specified path
+        /// in Azure Blob Storage.
+        /// </summary>
+        /// <param name="builder">The builder instance to modify.</param>
+        /// <param name="blobUri">The full URI where the key file should be stored.
+        /// The URI must contain the SAS token as a query string parameter.</param>
+        /// <returns>The value <paramref name="builder"/>.</returns>
+        /// <remarks>
+        /// The container referenced by <paramref name="blobUri"/> must already exist.
+        /// </remarks>
+        public static IDataProtectionBuilder PersistKeysToAzureBlobStorage(this IDataProtectionBuilder builder, Uri blobUri)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+            if (blobUri == null)
+            {
+                throw new ArgumentNullException(nameof(blobUri));
+            }
+
+            var uriBuilder = new BlobUriBuilder(blobUri);
+            BlobClient client;
+
+            // The SAS token is present in the query string.
+            if (uriBuilder.Sas == null)
+            {
+                client = new BlobClient(blobUri, new DefaultAzureCredential());
+            }
+            else
+            {
+                client = new BlobClient(blobUri);
+            }
+
+            return PersistKeystoAzureBlobStorageInternal(builder, client);
+        }
+
+        /// <summary>
+        /// Configures the data protection system to persist keys to the specified path
+        /// in Azure Blob Storage.
+        /// </summary>
+        /// <param name="builder">The builder instance to modify.</param>
+        /// <param name="blobUri">The full URI where the key file should be stored.
+        /// The URI must contain the SAS token as a query string parameter.</param>
+        /// <param name="tokenCredential">The credentials to connect to the blob.</param>
+        /// <returns>The value <paramref name="builder"/>.</returns>
+        /// <remarks>
+        /// The container referenced by <paramref name="blobUri"/> must already exist.
+        /// </remarks>
+        public static IDataProtectionBuilder PersistKeysToAzureBlobStorage(this IDataProtectionBuilder builder, Uri blobUri, TokenCredential tokenCredential)
```

Comment:
I find it a little confusing that we've got helpers for Sas/TokenCredential but not SharedKeyCredential/ConnectionString.  I get that users could instantiate it themselves and pass a BlobClient in to the other overload, but I'm not sure it will be as obvious to others.  I'm guessing this is a design decision to guide people toward better auth choices?  I think it's fine to Preview but would be worth looking to see if people struggle here.